### ISSUE TITLE
nvda will fail to close the pronounciation/symbol pronounciation dialog when the ok button is envoked.

### ISSUE THREAD

**Issue by kolohekai** [nvaccess/nvda#17344]
### Steps to reproduce:
1, cleanly install NVDA. uninstall NVDA completly from programs and features, and delete your user config in percent appdata percent to completely start over from scratch with nvda. 2. reinstall nvda in the normal way, preferably 2024.4 or 2024.4 rc 1 will do as well.
3, when NVDA is installed, go to the nvda menu with insert or capslock n then p for preferences. then go to the Punctuation/symbol pronunciation dialog.
4, once their, immidiately press the ok button, you will notice NVDA will not close the dialog and save your changes. the button will say pressed, but the dialog will stay open.
### Actual behavior:

NVDA+n
NVDA  menu
Settings...  s
down arrow
Speech dictionaries  subMenu  d
down arrow
Punctuation/symbol pronunciation...  p
enter
Symbol Pronunciation (English)  dialog
Symbols  list  Alt+  s
. sentence ending; Replacement: dot; Level: all; Preserve: always  1 of 4579
tab
Change selected symbol  grouping
Replacement  edit  Alt+  r  selected dot
tab
Level  combo box  all  collapsed  Alt+  l
tab
Send actual symbol to synthesizer  combo box  always  collapsed  Alt+  s
tab
Add  button  Alt+  a
tab
OK  button
pressed
pressed
tab
Cancel  button
pressed

### Expected behavior:

NVDA+n
NVDA  menu
p
Settings...  s
down arrow
Speech dictionaries  subMenu  d
down arrow
Punctuation/symbol pronunciation...  p
enter
Symbol Pronunciation (English)  dialog
Symbols  list  Alt+  s
. sentence ending; Replacement: dot; Level: all; Preserve: always  1 of 4579
tab
Change selected symbol  grouping
Replacement  edit  Alt+  r  selected dot
tab
Level  combo box  all  collapsed  Alt+  l
tab
Send actual symbol to synthesizer  combo box  always  collapsed  Alt+  s
tab
Add  button  Alt+  a
tab
OK  button
space
pressed
NVDA Speech Viewer
Show Speech Viewer on Startup  check box  not checked  Alt+  s
NVDA+4
speak command keys off

### NVDA logs, crash dumps and other attachments:
NVDA debug log is attached.
#### NVDA installed/portable/running from source:
NVDA Installed
#### NVDA version:
2024.4 rc1/2024.4
#### Windows version:
windows 11 24H2
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. older versions of NVDa from 2024.3 and older do not Have this behavior.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
[nvda-old wensday october 30, 2024.log](https://github.com/user-attachments/files/17581322/nvda-old.wensday.october.30.2024.log)

**Comment by SaschaCowley** [comment]
Looks like this was broken by #16823 . This is a regression in 2024.4.

**Comment by seanbudd** [comment]
Hi @nvdaes @LeonarddeR - could you look into #17346 which solves this issue?
